assets/src/js/frontend/give-donations.js

```diff
@@ -123,6 +123,22 @@ jQuery( function( $ ) {
 						$form.find( 'p#give-card-zip-wrap' ).removeClass( 'form-row-last' );
 					}
 
+					// Check whether the post code fields should be required
+					if (
+						'undefined' !== typeof ( response.zip_require ) &&
+						false === response.zip_require
+					) {
+						$form.find( 'input#card_zip' ).removeClass( 'required' );
+						$form.find( 'input#card_zip' ).attr( 'required', false );
+						$form.find( 'input#card_zip' ).attr( 'aria-required', false );
+						$form.find( 'label[for="card_zip"] span.give-required-indicator' ).addClass( 'give-hidden' );
+					} else {
+						$form.find( 'input#card_zip' ).addClass( 'required' );
+						$form.find( 'input#card_zip' ).attr( 'required', true );
+						$form.find( 'input#card_zip' ).attr( 'aria-required', true );
+						$form.find( 'label[for="card_zip"] span.give-required-indicator' ).removeClass( 'give-hidden' );
+					}
```

Comment:
This can be cleaned up to avoid redundant selectors:
```suggestion
					const zipRequired = !! response.zip_required;
					$form.find( 'input#card_zip' ).toggleClass( 'required', zipRequired );
					$form.find( 'input#card_zip' ).attr( 'required', zipRequired );
					$form.find( 'input#card_zip' ).attr( 'aria-required', zipRequired );
					$form.find( 'label[for="card_zip"] span.give-required-indicator' ).toggleClass( 'give-hidden', zipRequired );
```